src/main/scala/cognite/spark/v1/AssetsRelation.scala

```diff
@@ -69,8 +69,16 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
   }
 
   override def delete(rows: Seq[Row]): IO[Unit] = {
-    val deletes = rows.map(r => fromRow[DeleteItem](r))
-    deleteWithIgnoreUnknownIds(client.assets, deletes, config.ignoreUnknownIds)
+    val deletes = rows.map(fromRow[DeleteItem](_))
+
+    val ids = deletes.flatMap(_.id)
+    val externalIds = deletes.flatMap(_.externalId)
+
+    for {
+      _ <- deleteWithIgnoreUnknownIds(client.assets, ids, config.ignoreUnknownIds)
+      _ <- deleteByExternalIdsWithIgnoreUnknown(client.assets, externalIds, config.ignoreUnknownIds)
```

Comment:
Hmm, it would be nice to add support for delete by CogniteId into the sdk. We'd have to do half of the requests, and we'd get rid of some annoying duplicated logic

Reply:
it's not a bad idea, I make a PR in sdk :)
https://github.com/cognitedata/cognite-sdk-scala/pull/442/files